.github/workflows/build-test.yml

```diff
@@ -13,7 +12,24 @@ env:
   CI_NUM_THREADS: "16"
 
 jobs:
+  changes:
+    runs-on:  [ self-hosted, small ]
+    outputs:
+      source_code: ${{ steps.filter.outputs.source_code }}
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            source_code:
+              - 'bolt/**'
```

Comment:
I think we need one more path filter in case we touch anything in the `./scripts/conan` directory because recipe changes can affect behavior